x-pack/plugins/observability/server/lib/annotations/create_annotations_client.ts

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { APICaller, Logger } from 'kibana/server';
+import * as t from 'io-ts';
+import { SearchResponse, Client } from 'elasticsearch';
+import {
+  createAnnotationRt,
+  searchAnnotationsRt,
+  deleteAnnotationRt,
+  Annotation,
+  getAnnotationByIdRt,
+} from '../../../common/annotations';
+import { PromiseReturnType } from '../../../../apm/typings/common';
+import { createOrUpdateIndex } from '../../utils/create_or_update_index';
+
+type SearchParams = t.TypeOf<typeof searchAnnotationsRt>;
+type CreateParams = t.TypeOf<typeof createAnnotationRt>;
+type DeleteParams = t.TypeOf<typeof deleteAnnotationRt>;
+type GetByIdParams = t.TypeOf<typeof getAnnotationByIdRt>;
+
+interface IndexDocumentResponse {
+  _shards: {
+    total: number;
+    failed: number;
+    successful: number;
+  };
+  _index: string;
+  _type: string;
+  _id: string;
+  _version: number;
+  _seq_no: number;
+  _primary_term: number;
+  result: string;
+}
+
+export function createAnnotationsClient(params: {
+  index: string;
+  apiCaller: APICaller;
+  logger: Logger;
+}) {
+  const { index, apiCaller, logger } = params;
+
+  const initIndex = () =>
+    createOrUpdateIndex({
+      index,
+      mappings: {
+        dynamic: 'strict',
+        properties: {
+          annotation: {
+            properties: {
+              type: {
+                type: 'keyword',
+              },
+            },
+          },
+          message: {
+            type: 'text',
+          },
+          tags: {
+            type: 'keyword',
+          },
+          '@timestamp': {
+            type: 'date',
+          },
+          event: {
+            properties: {
+              created: {
+                type: 'date',
+              },
+            },
+          },
+          service: {
+            properties: {
+              name: {
+                type: 'keyword',
+              },
+              environment: {
+                type: 'keyword',
+              },
+              version: {
+                type: 'keyword',
+              },
+            },
+          },
+        },
+      },
```

Comment:
It's quite hard to read - perhaps extract `mappings`

Reply:
done